File: isma-hsm/src/main/java/ru/nstu/isma/core/hsm/events/HSMEventFunctionGroup.java
package ru.nstu.isma.core.hsm.events;

import ru.nstu.isma.core.hsm.exp.EXPOperator;
import ru.nstu.isma.core.hsm.exp.HMExpression;

import java.util.List;

/**
 * @author Maria
 * @since 16.05.2016
 */
public class HSMEventFunctionGroup {

    /** Список выражений с событийными функциями. */
    private final List<HMExpression> expressions;

    /**
     * Оператор, соединяющий событийные функции.
     * М.б. null, если в списке только 1 функция.
     */
    private final EXPOperator.Code operatorCode;

    public HSMEventFunctionGroup(List<HMExpression> expressions) throws IllegalArgumentException {
        int size = expressions.size();
        if (size != 1) {
            throw new IllegalArgumentException("Illegal size of the expressions: expected 1 but actual was " + size +
                    ". This constructor is only used for 1 event function.");
        }

        this.expressions = expressions;
        this.operatorCode = null;
    }

    public HSMEventFunctionGroup(List<HMExpression> expressions, EXPOperator.Code operatorCode) {
        int size = expressions.size();
        /*if (size < 2) {
            throw new IllegalArgumentException("Illegal size of the expressions: expected >= 2 but actual was " + size +
                    ". This constructor is only used for 2 or more event functions.");
        }*/
        this.expressions = expressions;

        /*if (operatorCode == EXPOperator.Code.AND ^ operatorCode == EXPOperator.Code.OR) {
            throw new IllegalArgumentException("Illegal operator: expected '&&' or '||', but actual was " + operatorCode + ".");
        }*/
        this.operatorCode = operatorCode;
    }

    public List<HMExpression> getExpressions() {
        return expressions;
    }

    public EXPOperator.Code getOperatorCode() {
        return operatorCode;
    }

}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/events/HSMEventFunctionGroup.kt
package ru.nstu.isma.hsm.events

import ru.nstu.isma.hsm.exp.EXPOperator
import ru.nstu.isma.hsm.exp.HMExpression

/**
 * @author Maria
 * @since 16.05.2016
 */
class HSMEventFunctionGroup {
    /** Список выражений с событийными функциями.  */
    val expressions: List<HMExpression?>

    /**
     * Оператор, соединяющий событийные функции.
     * М.б. null, если в списке только 1 функция.
     */
    val operatorCode: EXPOperator.Code?

    constructor(expressions: List<HMExpression?>) {
        val size = expressions.size
        require(size == 1) {
            "Illegal size of the expressions: expected 1 but actual was " + size +
                    ". This constructor is only used for 1 event function."
        }
        this.expressions = expressions
        operatorCode = null
    }

    constructor(expressions: List<HMExpression?>, operatorCode: EXPOperator.Code?) {
        val size = expressions.size
        /*if (size < 2) {
            throw new IllegalArgumentException("Illegal size of the expressions: expected >= 2 but actual was " + size +
                    ". This constructor is only used for 2 or more event functions.");
        }*/this.expressions = expressions

        /*if (operatorCode == EXPOperator.Code.AND ^ operatorCode == EXPOperator.Code.OR) {
            throw new IllegalArgumentException("Illegal operator: expected '&&' or '||', but actual was " + operatorCode + ".");
        }*/this.operatorCode = operatorCode
    }
}